### PR TITLE
WIP - proof of concept; staking to start work on a bounty

### PR DESCRIPTION
**Pull request by owocki**
##### Description

Allows a 'stake to apply work' permission type; and the subsequent logic to allow the user to stake (right now, using tips architecture). 

Still TODO - the ability to take away the users stake (just reassigning the tip), and the communications for that to the user.

##### Refers/Fixes

Self, an idea we've had forever to increase contributor quality (I can't find a ticket for it tho)

#1893 

##### Testing

# submit a bounty
Screen Recording 2019-08-05 at 07 51 AM

# start work

Screen Recording 2019-08-05 at 07 53 AM

### CODE REVIEW

File: app/dashboard/templates/addinterest.html [hunk 1]
```diff
@@ -23,7 +23,20 @@
     <div class="col-12">
       <h5 class="text-center font-title">{% trans "Submit a Plan" %}</h5>
     </div>
-    {% if user_logged_in %}
+    {% if stake_required %}
+      <div class="col-12 text-center">
+        <div class="stake">
+          <div class="stake__text">
+            The bounty funder has specified that you must stake ETH to work on this bounty.  Staking ETH is a sign that you are taking the bounty seriously.  If you successfully get this bounty done, you will receive your ETH back.
```
Comment (octavioamu) [inline app/dashboard/templates/addinterest.html]: I for any reason I need to stop working I receive it back?

File: app/dashboard/templates/shared/issue_details.html [hunk 2]
```diff
@@ -101,6 +101,16 @@
       </span>
     </label>
   </div>
+  <div class="col-12 col-md-4 btn-group-toggle">
+    <label class="btn btn-block btn-radio permissionless my-2 font-weight-bold py-3 px-4 text-left">
+      <input type="radio" name="permission_type" id="stake_required" value="stake_required" autocomplete="off">
+      {% trans "Stake Required" %}
+      <span class="d-block font-weight-normal">
+        <i class="fas fa-user-cog "></i>
+        {% trans "Any bounty hunter who wants to work on the issue will have to stake 10% of the value of the issue." %}
```
Comment (octavioamu) [inline app/dashboard/templates/shared/issue_details.html]: maybe we can add a tooltip to include this full explanation next {% trans "Stake Required" %} and change the one displayed by a short version
`{% trans "Stake Required" %} <i class="far fa-info" data-toggle="tooltip" title="Any bounty hunter who wants to work on the issue will have to stake 10% of the value of the issue."></i> `

File: app/dashboard/templates/addinterest.html (hunk 1 above)
Comment (owocki) [reply app/dashboard/templates/addinterest.html]: we should figure out the triggers... i think that if you abandon the bounty then you should NOT get the stake back.

**Comment by thelostone-mc** [comment]
Closing this out as this would work with cross-chain integration and that would need a lil bit more time and effort

**Pull request closed without merging**